src/SSHDebugPS/UI/ViewModels/ContainerViewModel.cs

```diff
@@ -0,0 +1,180 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.ComponentModel;
+using liblinux.Shell;
+using Microsoft.SSHDebugPS.Docker;
+using WindowsInput = System.Windows.Input;
+
+namespace Microsoft.SSHDebugPS.UI
+{
+    public interface IContainerViewModel
+        : INotifyPropertyChanged, IEquatable<IContainerViewModel>
+    {
+        string Name { get; }
+        string Id { get; }
+        WindowsInput.ICommand ExpandCommand { get; }
+        bool GetResult(out string selectedQualifier);
+        bool IsExpanded { get; set; }
+        bool IsSelected { get; set; }
+    }
+
+    public abstract class ContainerViewModel<T> : IContainerViewModel
```

Comment:
><T> [](start = 44, length = 3)

Out of curiosity - are you getting anything out of the fact that this is generic? It seems like you could just replace `T` with `IContainerInstance`, but maybe I am missing something.

Reply:
It was so I can access concrete type info in derived classes without having to cast it.